src/App.jsx

```diff
@@ -14,45 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const sortBy = [
+  { name: 'Sort alphabetically', color: 'is-info' },
+  { name: 'Sort by length', color: 'is-success' },
+  { name: 'Reverse', color: 'is-warning' },
+  { name: 'Reset', color: 'is-danger' },
+];
+
+function sortGoods(goods, type, reverse) {
+  const goodsSorted = [...goods];
+
+  if (type) {
+    goodsSorted.sort((good1, good2) => (
+      (type === 'Sort alphabetically')
+        ? good1.localeCompare(good2)
+        : good1.length - good2.length));
+  }
+
+  if (reverse) {
+    goodsSorted.reverse();
+  }
+
+  return goodsSorted;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortType, setSortType] = useState(false);
+  const [isReversed, setisReversed] = useState(false);
+  const goods = sortGoods(goodsFromServer, sortType, isReversed);
+
+  function isLight(name) {
```

Comment:
[Checklist](https://github.com/mate-academy/react_list-of-goods-js/blob/master/checklist.md) p.4